pkg/manifest/init.go

```diff
@@ -141,28 +158,70 @@ func CreateResources(prefix, gitOpsRepo, gitOpsWebhook string) (map[string]inter
 	outputs[rolesPath] = roles.CreateClusterRole(meta.NamespacedName("", roles.ClusterRoleName), Rules)
 
 	sa := roles.CreateServiceAccount(meta.NamespacedName(cicdNamespace, saName))
+
+	if dockerConfigJSONPath != "" {
+		dockerSecret, err := CreateDockerSecret(dockerConfigJSONPath, cicdNamespace)
+		if err != nil {
+			return nil, err
+		}
+		outputs[dockerConfigPath] = dockerSecret
+
+		// add secret and sa to outputs
+		outputs[serviceAccountPath] = roles.AddSecretToSA(sa, dockerSecretName)
+	}
+
 	outputs[rolebindingsPath] = roles.CreateRoleBinding(meta.NamespacedName(cicdNamespace, roleBindingName), sa, "ClusterRole", roles.ClusterRoleName)
 
 	outputs[tasksPath] = tasks.CreateDeployFromSourceTask(cicdNamespace, GetPipelinesDir("", prefix))
 	outputs[ciPipelinesPath] = pipelines.CreateCIPipeline(meta.NamespacedName(cicdNamespace, "ci-dryrun-from-pr-pipeline"), cicdNamespace)
 	outputs[cdPipelinesPath] = pipelines.CreateCDPipeline(meta.NamespacedName(cicdNamespace, "cd-deploy-from-push-pipeline"), cicdNamespace)
+	outputs[appCiPipelinesPath] = pipelines.CreateCIPipeline(meta.NamespacedName(cicdNamespace, "app-ci-pipeline"), cicdNamespace)
```

Comment:
Pipelines are repeated here. We need to generate separate pipelines that'll deploy the app to a given environment